lib/sms/sms_submit.c

```diff
@@ -88,6 +104,102 @@ static int sms_submit_encode_number(
 	return 0;
 }
 
+/**
+ * @brief Encode SMS Service Center Address (SCA) field.
+ *
+ * @details SCA field is used in SMS-SUBMIT message. This is specified piece by piece
+ * in the following specifications:
+ *   - 3GPP TS 23.040 chapter 9.1.2.5
+ *   - 3GPP TS 27.005 chapter 3.1 (see <pdu>, <tosca> and <toda>)
+ *   - 3GPP TS 27.005 chapter 3.3.1
+ *   - 3GPP TS 27.005 chapter 4.3
+ *
+ * @param[out] sca_field Buffer for SCA field.
+ *                       Must be at least 23 bytes long to include the following information:
+ *                       - Address/number: 20
+ *                       - Address length: 1
+ *                       - Type of address: 1
+ *                       - NUL termination: 1
```

Comment:
nit: 
```suggestion
 *                       - NULL termination: 1
```
Here and others

Reply:
Hmmm, NULL and NUL termination is used in a mixed manner. Google tells that NUL is the name of the '\0' character and hence probably more correct here although there are reasons why you would call it NULL termination.